src/charm.py

```diff
@@ -70,7 +72,7 @@ def get_listener(cluster: str, port: int):
                 match=api.route.RouteMatch(prefix="/"),
                 route=api.route.RouteAction(
                     cluster=cluster,
-                    max_grpc_timeout=timedelta(seconds=0),
+                    max_grpc_timeout=timedelta(seconds=60),
```

Comment:
According to the docs, this is:

> Deprecated by [grpc_timeout_header_max](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto.html#envoy-v3-api-field-config-route-v3-routeaction-maxstreamduration-grpc-timeout-header-max) If present, and the request is a gRPC request, use the [grpc-timeout header](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md), or its default value (infinity) instead of [timeout](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto.html#envoy-v3-api-field-config-route-v3-routeaction-timeout), but limit the applied timeout to the maximum value specified here.

Should we use that instead?

Reply:
I used this field in order to not diverge from [upstream](https://github.com/kubeflow/pipelines/blob/master/third_party/metadata_envoy/envoy.yaml#L26).